GPU/Common/TextureShaderCommon.cpp

```diff
@@ -236,7 +236,7 @@ Draw2DPipeline *TextureShaderCache::GetDepalettizeShader(uint32_t clutMode, GETe
 std::vector<std::string> TextureShaderCache::DebugGetShaderIDs(DebugShaderType type) {
 	std::vector<std::string> ids;
 	for (auto &iter : depalCache_) {
-		ids.push_back(StringFromFormat("%08x", iter.first));
```

Comment:
How does this even compile?? You're pushing a compile-time string literal + the iterator into a vector of `std::strings`!

Reply:
> How does this even compile?? You're pushing a compile-time string literal + the iterator into a vector of `std::strings`!

ha ha ha, you're really right, it's amazing how my compiler didn't mention this, as well as all the others. It's good that all contributors can keep track where and how errors are hidden when reviewing code. It's pity that UB is so impossible to identify.

Reply:
Ok, why did I say it's the iterator...? It's `iter.first`, not `iter`. That makes more sense - the compiler used some valid `std::string` constructor, probably this one:

![image](https://github.com/hrydgard/ppsspp/assets/62447396/30b8c8f5-a021-4e8f-8723-bd0a2d09823d)


And no, it's not UB, that's a logic error.